components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java

```diff
@@ -220,6 +233,15 @@ protected void deactivate(ComponentContext ctxt) {
         if (Objects.nonNull(transactionCountExecutor)) {
             transactionCountExecutor.shutdownNow();
         }
+        List handlers = serverManager.getServerContextInformation().getSynapseEnvironment().getSynapseHandlers();
+        Iterator<SynapseHandler> iterator = handlers.iterator();
+        while (iterator.hasNext()) {
+            SynapseHandler handler = iterator.next();
+            if (((handler instanceof AbstractExtendedSynapseHandler)) &&
+                    (!((AbstractExtendedSynapseHandler) handler).handleServerShutDown())) {
+                    return;
```

Comment:
If we return here, the rest of the steps will not execute and the server will not shutdown.

Reply:
Logged the warning